akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -127,17 +127,22 @@ final class BootstrapSetup private (
     val actorRefProvider: Option[ProviderSelection] = None)
     extends Setup {
 
-  def withClassloader(classLoader: ClassLoader): BootstrapSetup =
-    new BootstrapSetup(Some(classLoader), config, defaultExecutionContext, actorRefProvider)
+  def withClassloader(classLoader: ClassLoader): BootstrapSetup = copy(classLoader = Some(classLoader))
 
-  def withConfig(config: Config): BootstrapSetup =
-    new BootstrapSetup(classLoader, Some(config), defaultExecutionContext, actorRefProvider)
+  def withConfig(config: Config): BootstrapSetup = copy(config = Some(config))
 
   def withDefaultExecutionContext(executionContext: ExecutionContext): BootstrapSetup =
-    new BootstrapSetup(classLoader, config, Some(executionContext), actorRefProvider)
+    copy(defaultExecutionContext = Some(executionContext))
```

Comment:
how does this work? is this only replacing the default-dispatcher?

I think it's important to be able to pass in one single ExecutionContext that is used as both default-dispatcher and internal-dispatcher. Use case is resource constrained environments. Maybe that is already possible here, by defining `internal-dispatcher="akka.actor.default-dispatcher"`? Might worth trying out and add a section to the dispatchers documentation.

Reply:
I tried to provide external execution context for both default and internal but then backed out of it decided to try to do that in a separate go, this is just a minor cleanup that I left from that.

Reply:
ok, but wouldn't it be solved by config `internal-dispatcher="akka.actor.default-dispatcher"`.
I don't think external exectionContext is used much so slightly inconvenient (but documented) way would be ok.

Using external exectionContext for internal but not default dispatcher seems like an odd case that we don't have to support.

Reply:
I'll verify and document that you can use a pre-existing executor when closer to done with this!